src/persist/src/file.rs

```diff
@@ -419,6 +419,10 @@ impl Blob for FileBlob {
             if let Some(name) = file_name {
                 let name = name.to_str();
                 if let Some(name) = name {
+                    // Exclude the lockfile from the user's list of keys.
```

Comment:
This could be `if name == Some(Self::LOCKFILE_PATH)`

Reply:
I might be missing something, but I think this needs to unwrap the option separate from checking whether it matches the lockfile because we use the `name` bound in `if let Some(name) = name {` on line 426.

More concretely, to do it the way you suggest I think I have to do something like this, or something more verbose?

```
diff --git a/src/persist/src/file.rs b/src/persist/src/file.rs
index 70d980455..303aef50b 100644
--- a/src/persist/src/file.rs
+++ b/src/persist/src/file.rs
@@ -418,13 +418,12 @@ impl Blob for FileBlob {
             let file_name = path.file_name();
             if let Some(name) = file_name {
                 let name = name.to_str();
-                if let Some(name) = name {
+                if let Some(Self::LOCKFILE_PATH) = name {
                     // Exclude the lockfile from the user's list of keys.
-                    if name == Self::LOCKFILE_PATH {
-                        continue;
-                    }
-                    ret.push(name.to_owned());
+                    continue;
                 }
+
+                ret.push(name.expect("known to be valid utf-8").to_string());
             }
         }
```

Of course, let me know if I'm being silly and missing something

Reply:
no, you're right. sorry for the silly suggestion :sweat_smile: